src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int COUNT_OF_FIGURE = 5; //Count of different figure classes
+    private Random random;
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier() {
+        this.random = new Random();
+        this.colorSupplier = new ColorSupplier();
+    }
+
+    public Figure getRandomFigure() {
+        int bound = 1000;
+        /*
+        typeOfFigure : 0 - IsoscelesTrapezoid,
+        1 - Square,
+        2 - RightTriangle,
+        3 - Circle,
+        4 - Rectangle
+         */
+        int typeOfFigure = random.nextInt(COUNT_OF_FIGURE);
+        switch (typeOfFigure) {
+            case 0 :
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(bound), random.nextInt(bound), random.nextInt(bound));
+            case 1 :
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(bound));
+            case 2 :
+                return new RightTriangle(colorSupplier.getRandomColor(), random.nextInt(bound));
+            case 3 :
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(bound));
+            case 4 :
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(bound), random.nextInt(bound));
+            default:
+                return new Square(Color.BLACK, 10);
```

Comment:
```suggestion
            default :
                return getRandomRectangle();
```
Create methods like getRandomRectangle(); for each case to improve code readability